Cannot compile with GCC 15.1.1

I currently cannot compile the source code, as there is some issue with compiling with GCC 15.1.1.

```
-- The C compiler identification is GNU 15.1.1
-- The CXX compiler identification is GNU 15.1.1
```

The error itself seems to originate from our usage of `sol2`

```
In file included from /XXX/ja2-stracciatella/src/externalized/scripting/ScriptingExtensionsLua.cc:7:
/XXX/ja2-stracciatella/_bin/lib-sol2/src/single/include/sol/sol.hpp: In member function ‘T& sol::optional<T&>::emplace(Args&& ...)’:
/XXX/ja2-stracciatella/_bin/lib-sol2/src/single/include/sol/sol.hpp:6059:31: error: ‘class sol::optional<T&>’ has no member named ‘construct’ [-Wtemplate-body]
 6059 |                         this->construct(std::forward<Args>(args)...);
      |                               ^~~~~~~~~
make[2]: *** [CMakeFiles/ja2.dir/build.make:317: CMakeFiles/ja2.dir/src/externalized/scripting/ScriptingExtensionsLua.cc.o] Error 1
```

Seems like CLang 19.1 has the same issue: https://github.com/ThePhD/sol2/issues/1617

@misanthropos I've seen that you applied some patches to the Gentoo sol2 version which fix those issues. Do you know which combination of patches is needed to make it work? Then we could apply them to our version of `sol2` as well.

--- a/dependencies/lib-sol2/CMakeLists.txt
+++ b/dependencies/lib-sol2/CMakeLists.txt
@@ -26,6 +26,13 @@ configure_file(
     @ONLY
 )
 
+# copy patch to build dir
+configure_file(
+    "${CMAKE_CURRENT_SOURCE_DIR}/getter/gcc-15.patch"
+    "${CMAKE_CURRENT_BINARY_DIR}/getter/gcc-15.patch"
+    @ONLY
+)
+
 # execute getter
 execute_process(COMMAND ${CMAKE_COMMAND} . "-G${CMAKE_GENERATOR}" "-DCMAKE_SYSTEM_VERSION=${CMAKE_SYSTEM_VERSION}" WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/getter")
 execute_process(COMMAND ${CMAKE_COMMAND} --build "${CMAKE_CURRENT_BINARY_DIR}/getter")

--- a/dependencies/lib-sol2/getter/CMakeLists.txt.in
+++ b/dependencies/lib-sol2/getter/CMakeLists.txt.in
@@ -22,4 +22,6 @@ externalproject_add(get-sol2
         BUILD_COMMAND       ""
         INSTALL_COMMAND     ""
         TEST_COMMAND        ""
+    # called several times, so don't error out when a patched state is detected
+    PATCH_COMMAND grep -q std::in_place, single/include/sol/sol.hpp || { sed -i s,\\r,, single/include/sol/sol.hpp && patch -p3 < ${CMAKE_CURRENT_SOURCE_DIR}/gcc-15.patch; }
 )

--- a/dependencies/lib-sol2/getter/gcc-15.patch
+++ b/dependencies/lib-sol2/getter/gcc-15.patch
@@ -1,0 +1,40 @@
+--- ./lib-sol2/src/single/include/sol/sol.hpp.orig	2025-07-26 14:05:48.679789792 +0200
++++ ./lib-sol2/src/single/include/sol/sol.hpp	2025-07-26 14:45:46.719189562 +0200
+@@ -5150,7 +5150,7 @@
+ 			static_assert(std::is_constructible<T, Args&&...>::value, "T must be constructible with Args");
+ 
+ 			*this = nullopt;
+-			this->construct(std::forward<Args>(args)...);
++			new (static_cast<void*>(this)) optional(std::in_place, std::forward<Args>(args)...);
+ 			return value();
+ 		}
+ 
+@@ -6056,7 +6056,8 @@
+ 			static_assert(std::is_constructible<T, Args&&...>::value, "T must be constructible with Args");
+ 
+ 			*this = nullopt;
+-			this->construct(std::forward<Args>(args)...);
++			new (static_cast<void*>(this)) optional(std::in_place, std::forward<Args>(args)...);
++			return **this;
+ 		}
+ 
+ 		/// Swaps this optional with the other.
+@@ -14685,7 +14686,17 @@
+ 						lua_getglobal(L, &key[0]);
+ 					}
+ 					else {
+-						lua_getfield(L, tableindex, &key[0]);
++						if constexpr (std::is_same_v<std::decay_t<Key>, const char*>) {
++							// Handle const char* case
++							if (key != nullptr) {
++								lua_getfield(L, tableindex, key);
++							} else {
++								push(L, lua_nil);
++							}
++						} else {
++							// Handle std::string case
++							lua_getfield(L, tableindex, key.c_str());
++						}
+ 					}
+ 				}
+ #if SOL_LUA_VESION_I_ >= 503

